lib/tool_shed/galaxy_install/installed_repository_manager.py

```diff
@@ -42,6 +42,13 @@ def __init__(self, app):
         self.tool_configs = self.app.config.tool_configs
         if self.app.config.migrated_tools_config not in self.tool_configs:
             self.tool_configs.append(self.app.config.migrated_tools_config)
+
+        self.tool_trees = []
+        for tool_config in self.tool_configs:
+            tree, error_message = xml_util.parse_xml(tool_config)
+            log.error(error_message)
```

Comment:
```python
            if error_message:
                log.error(error_message)
```

Reply:
Could we just store those `tool_path` attributes ? Seems that is the only thing needed here.

Reply:
+1 to mvdbeek's suggestion.  And, to be certain, these aren't modified after startup, correct?

Reply:
An admin could do this by hand, but those shouldn't be changed in normal operation.